src/slices/filters.ts

```diff
@@ -47,6 +57,34 @@ const reducers = {
       });
     });
   },
+  toggleFilterOption: (state: FiltersState, { payload }: PayloadAction<FilterPayload>) => {
+    if (!state.static) {
+      state.static = {};
+    }
+    const { filterCollectionId, filter: targetFilter, shouldSelect } = payload;
+    if (!filterCollectionId) {
+      console.warn(`invalid static filters id: ${filterCollectionId}`);
+      return;
+    }
+    const filterCollection = state.static[filterCollectionId];
+    const foundFilter = filterCollection?.find(storedSelectableFilter => {
+      const storedFilter = storedSelectableFilter.filter;
+      return storedFilter.fieldId === targetFilter.fieldId
+        && storedFilter.matcher === targetFilter.matcher
+        && storedFilter.value === targetFilter.value;
+    });
+    if (foundFilter) {
+      foundFilter.selected = shouldSelect;
+    } else if (shouldSelect) {
+      const selectedFilter = { filter: targetFilter, selected: shouldSelect };
+      filterCollection
+        ? filterCollection.push(selectedFilter)
+        : state.static[filterCollectionId] = [selectedFilter];
+    } else {
+      console.warn('Could not unselect a non-existing filter option in state'
```

Comment:
It might be helpful to include the filterCollectionId here?

Reply:
good point, updated!